dbt/include/bigquery/macros/utils/get_columns_spec_ddl.sql

```diff
@@ -1,18 +1,29 @@
 {% macro bigquery__get_columns_spec_ddl() %}
   {# loop through user_provided_columns to create DDL with data types and constraints #}
-    {%- set ns = namespace(at_least_one_check=False) -%}
+    {%- set ns = namespace(at_least_one_check=False, at_least_one_pk=False) -%}
     {%- set user_provided_columns = model['columns'] -%}
     (
-    {% for i in user_provided_columns -%}
+    {% for i in user_provided_columns %}
       {%- set col = user_provided_columns[i] -%}
-      {% set constraints = col['constraints'] -%}
-      {%- set ns.at_least_one_check = ns.at_least_one_check or col['constraints_check'] %}
-      {{ col['name'] }} {{ col['data_type'] }} {% for x in constraints %} {{ x or "" }} {% endfor %} {{ "," if not loop.last }}
-    {%- endfor %}
-  )
+      {%- set constraints = col['constraints'] -%}
+      {{ col['name'] }} {{ col['data_type'] }}
+      {%- for c in constraints -%}
+        {%- if c.type == "check" -%}
+          {%- set ns.at_least_one_check = True -%}
+        {%- elif c.type == "primary_key" -%}
+          {%- set ns.at_least_one_pk = True -%}
+        {%- else %} {{ adapter.render_raw_column_constraint(c) }}
+        {%- endif -%}
+      {%- endfor -%}
+      {{ "," if not loop.last }}
+    {% endfor -%}
+    )
   {%- if ns.at_least_one_check -%}
```

Comment:
Could we also move the exception handling into `adapter.render_raw_column_constraints()`? That would allow the jinja work to focus on templating.

Reply:
I agree that we should move that logic into the adapter's Python code. I think it is likely that @emmyoop will do so while working on https://github.com/dbt-labs/dbt-core/issues/7067, and we should be sure to remove all of the Jinja that deals with warnings as part of that work.

Reply:
That's right, @MichelleArk did mention that we were approaching that more as one big refactor. Makes sense to me.